samza-core/src/main/java/org/apache/samza/storage/blobstore/index/DirIndex.java

```diff
@@ -48,6 +51,12 @@ public DirIndex(String dirName,
     Preconditions.checkNotNull(filesRemoved);
     Preconditions.checkNotNull(subDirsPresent);
     Preconditions.checkNotNull(subDirsRemoved);
+    // check to validate that a file is not present in file removed and file present list
```

Comment:
When could this happen? What scenario are we defending against?

Nit: s/check/Check

Reply:
This is basic sanity check, similar to what we are doing in [DirDiff](https://github.com/apache/samza/blob/master/samza-core/src/main/java/org/apache/samza/storage/blobstore/diff/DirDiff.java#L84).